stores/dotenv/store.go

```diff
@@ -0,0 +1,152 @@
+package dotenv //import "go.mozilla.org/sops/stores/dotenv"
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/base64"
+	"encoding/gob"
+	"fmt"
+	"go.mozilla.org/sops"
+	"go.mozilla.org/sops/stores"
+	"strings"
+)
+
+const SopsPrefix = "sops_"
+
+// Store handles storage of dotenv data
+type Store struct {
+}
+
+func (store *Store) LoadEncryptedFile(in []byte) (sops.Tree, error) {
+	branch, err := store.LoadPlainFile(in)
+	if err != nil { return sops.Tree{}, err }
+
+	resultBranch := make(sops.TreeBranch, 0)
+	metadata := stores.Metadata{}
+	md_found := false
+	for _, item := range branch {
+		// FIXME: use sops_* items instead
+		if strings.HasPrefix(item.Key.(string), SopsPrefix) {
+			if item.Key == SopsPrefix + "metadata" {
+				metadata, err = FromGOB64(fmt.Sprint(item.Value))
+				if err != nil { return sops.Tree{}, err }
+				md_found = true
+				break
+			}
+		} else {
+			resultBranch = append(resultBranch, item)
+		}
+	}
+	if !md_found { return sops.Tree{}, sops.MetadataNotFound }
+
+	internalMetadata, err := metadata.ToInternal()
+	if err != nil { return sops.Tree{}, err }
+
+	return sops.Tree{
+		Branch:   resultBranch,
+		Metadata: internalMetadata,
+	}, nil
+}
+
+func (store *Store) LoadPlainFile(in []byte) (sops.TreeBranch, error) {
+	branch := make(sops.TreeBranch, 0)
+	reader := bytes.NewReader(in)
```

Comment:
[`bytes.Split`](https://golang.org/pkg/bytes/#Split) should work here as well, instead of needing to use buffered IO.

Simple example: https://play.golang.org/p/NpP1IcidSIt